io-engine/src/lvs/lvs_snapshot.rs

```diff
@@ -0,0 +1,74 @@
+use super::Error;
+use crate::lvs::{LogicalVolume, Lvol, Lvs};
+use async_trait::async_trait;
+use spdk_rs::libspdk::spdk_blob_is_read_only;
+
+pub struct LvolSnapshot {
+    logical_vol: Lvol,
+    group_id: Option<String>,
+    entity_id: Option<String>,
+    parent: Option<String>,
+}
+
+#[async_trait(? Send)]
+impl LogicalVolume for LvolSnapshot {
+    /// returns the name of the Snapshot
+    fn name(&self) -> String {
+        self.logical_vol.as_bdev().name().to_string()
+    }
+
+    /// returns the UUID of the Snapshot
+    fn uuid(&self) -> String {
+        self.logical_vol.as_bdev().uuid_as_string()
+    }
+    /// return lvs for the Logical Volume
+    fn lvs(&self) -> Lvs {
+        Lvs::from_inner_ptr(self.logical_vol.as_inner_ref().lvol_store)
+    }
+
+    /// returns the pool name of the Snapshot
+    fn pool_name(&self) -> String {
+        self.logical_vol.lvs().name().to_string()
+    }
+
+    /// returns the pool uuid of the Snapshot
+    fn pool_uuid(&self) -> String {
+        self.logical_vol.lvs().uuid()
+    }
+
+    ///  wipe the first 8MB if unmap is not supported on failure the operation
+    /// needs to be repeated
+    async fn wipe_super(&self) -> Result<(), Error> {
+        Ok(())
+    }
+
+    /// returns a boolean indicating if the Logical Volume is thin provisioned
+    fn is_thin(&self) -> bool {
+        self.logical_vol.as_inner_ref().thin_provision
+    }
+
+    /// returns a boolean indicating if the Logical Volume is read-only
+    fn is_read_only(&self) -> bool {
+        unsafe { spdk_blob_is_read_only(self.logical_vol.blob_checked()) }
+    }
+
+    /// return the size of the Snapshot in bytes
+    fn size(&self) -> u64 {
+        self.logical_vol.as_bdev().size_in_bytes()
+    }
+
+    /// Get Entity Id of the Snapshot
+    fn get_entity_id(&self) -> String {
+        self.entity_id.clone().unwrap_or_default()
+    }
+
+    /// Get Group Id of the Snapshot
+    fn group_id(&self) -> String {
+        self.group_id.clone().unwrap_or_default()
+    }
+
+    /// Get Parent Id of the Snapshot
+    fn get_parent_id(&self) -> String {
```

Comment:
It would also be nicer to make this function return Option<String>, as for the first snapshot there is no parent snapshot in the snapshot chain. Thanks.

Reply:
3 Snapshot Related parameters functions return is chanegd to Option.